### ISSUE TITLE
[BUG]: False positive for "The value "foo" of the 'for' attribute is not the 'id' of a valid <input> element"

### ISSUE THREAD

**Issue by irv** [IBMa/equal-access#2068]
### Project

accessibility-checker-engine

### Browser

Chrome, Safari, Firefox

### Operating system

MacOS

### Description

I guess related to PR #1944

When testing against a select eleemnt, a false positive is displayed for this rule.

But WCAG 2.2 H44 states that this is not just applied to `<input>` elements, but also `<select>` and `<textarea>`

https://www.w3.org/WAI/WCAG22/Techniques/html/H44

### Steps to reproduce

```html
<label for="my_select">A Selection</label>
<select id="my_select">
<option value="1">One</option>
<option value="2">Two</option>
<option value="3">Three</option>
</select>
```

**Comment by shunguoy** [comment]
@irv, I ran your test example, but I don't see the error. Can you please check again or post a test code with the error? Thanx.

**Comment by tombrunet** [comment]
It looks like we need to tweak the message because it indicates that it must be an `input` even though it already does the right thing for other element types. The example that you provide does not trigger the indicated error by itself. Usually this happens because you have the same id on another element that isn't appropriate, so even though you think you'd associated a label to the element you intended, you've actually associated it with some other element that isn't correct.

**Comment by shunguoy** [comment]
The change in the messages was made and will be deployed in the Nov. release.

**Comment by irv** [comment]
_right_, OK. I think in my case it's not that it's a duplicate id, but that the element is hidden in the dom by something
```html
<label for="my_select">A Selection</label>
<select id="my_select" hidden="">
<option value="1">One</option>
<option value="2">Two</option>
<option value="3">Three</option>
</select>
```

So that triggers it, for example. Agree updating the text wouldbe very helpful

**Comment by shunguoy** [comment]
@irv Yes, your test code triggers the error, but the error message is unclear. The violation message was updated for all the labelable elements, including <input>, <select>, etc.   The change will be deployed in the Nov. release.  I am going to close the issue.
